grouper/permissions.py

```diff
@@ -146,23 +123,23 @@ def grant_permission(session, group_id, permission_id, argument=""):
     session.commit()
 
 
-def grant_permission_to_service_account(session, account, permission, argument=""):
-    """
-    Grant a permission to this service account. This will fail if the (permission, argument) has
-    already been granted to this group.
+def grant_permission_to_service_account(
+    session: Session, account: ServiceAccount, permission: Permission, argument: str = ""
+) -> None:
+    """Grant a permission to this service account.
+
+    This will fail if the (permission, argument) has already been granted to this group.
 
     Args:
-        session(models.base.session.Session): database session
-        account(ServiceAccount): a ServiceAccount object being granted a permission
-        permission(Permission): a Permission object being granted
-        argument(str): must match constants.ARGUMENT_VALIDATION
+        session: Database session
+        account: A ServiceAccount object being granted a permission
+        permission: A Permission object being granted
+        argument: Must match constants.ARGUMENT_VALIDATION
 
     Throws:
         AssertError if argument does not match ARGUMENT_VALIDATION regex
     """
-    assert re.match(
-        ARGUMENT_VALIDATION + r"$", argument
-    ), "Permission argument does not match regex."
+    assert re.match(ARGUMENT_VALIDATION + r"$", argument), "Invalid permission argument"
```

Comment:
Should have a `r"^"` at the start as well to be totally comprehensive, right?

Reply:
`re.match` is implicitly anchored at the start of the string (as opposed to `re.search`, which is not). I find this confusing in Python as well.